live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/messaging/in/RepoEventMapperTest.java

```diff
@@ -0,0 +1,196 @@
+    private static final Set<String> NODE_ASPECT_NAMES = Set.of(
+        "cm:titled",
+        "cm:auditable");
+    private static final boolean NODE_IS_FOLDER = true;
+    private static final boolean NODE_IS_FILE = false;
+    private static final long NODE_CREATED_AT = 1_690_000_000_050L;
+    private static final NodeProperty<String> NODE_TITLE = new NodeProperty<>("cm:title", "some title");
+    private static final Set<NodeProperty<?>> NODE_PROPERTIES = Set.of(NODE_TITLE);
+    private final CamelEventMapper camelEventMapper = new CamelEventMapper(ObjectMapperFactory.createInstance());
+    private final RepoEventMapper repoEventMapper = new RepoEventMapper();
+
+    @Test
+    void mapToIngestNewNodeEvent()
+    {
+        // given
+        RepoEvent<DataAttributes<NodeResource>> event = getEvent("node-created-event.json");
+
+        Node node = new Node(
+            NODE_ID,
+            NODE_NAME,
+            NODE_PRIMARY_ASSOC_Q_NAME,
+            NODE_TYPE,
+            NODE_CREATED_BY_USER_WITH_ID,
+            NODE_MODIFIED_BY_USER_WITH_ID,
+            NODE_ASPECT_NAMES,
+            NODE_IS_FILE,
+            NODE_IS_FOLDER,
+            NODE_CREATED_AT,
+            NODE_PROPERTIES);
+
+        IngestNewNodeEvent expectedEvent = new IngestNewNodeEvent(
+            EVENT_TIMESTAMP,
+            node);
+
+        // when
+        IngestNewNodeEvent actualEvent = repoEventMapper.mapToIngestNewNodeEvent(event);
+
+        // then
+        assertEquals(expectedEvent, actualEvent);
+    }
+
+    @Test
+    void mapToIngestNewNodeEvent2()
+    {
+        // given
+        RepoEvent<DataAttributes<NodeResource>> event = mock();
+        DataAttributes<NodeResource> data = mock();
+        NodeResource nodeResource = mock();
+
+        UserInfo nodeCreator = mockUser(NODE_CREATED_BY_USER_WITH_ID);
+        UserInfo nodeModifier = mockUser(NODE_MODIFIED_BY_USER_WITH_ID);
+
+        when(event.getTime()).thenReturn(dateFromTimestamp(EVENT_TIMESTAMP));
+        when(event.getData()).thenReturn(data);
+        when(data.getResource()).thenReturn(nodeResource);
+        when(nodeResource.getId()).thenReturn(NODE_ID);
+        when(nodeResource.getName()).thenReturn(NODE_NAME);
+        when(nodeResource.getPrimaryAssocQName()).thenReturn(NODE_PRIMARY_ASSOC_Q_NAME);
+        when(nodeResource.getNodeType()).thenReturn(NODE_TYPE);
+        when(nodeResource.getCreatedByUser()).thenReturn(nodeCreator);
+        when(nodeResource.getModifiedByUser()).thenReturn(nodeModifier);
+        when(nodeResource.getAspectNames()).thenReturn(NODE_ASPECT_NAMES);
+        when(nodeResource.isFile()).thenReturn(NODE_IS_FILE);
+        when(nodeResource.isFolder()).thenReturn(NODE_IS_FOLDER);
+        when(nodeResource.getCreatedAt()).thenReturn(dateFromTimestamp(NODE_CREATED_AT));
+        when(nodeResource.getProperties()).thenReturn(createPropertiesMap("cm:title", "some title", "cm:description", null));
```

Comment:
Could we simplify this by returning a real NodeResource from the mock?

Reply:
yep, we can actually do that - should I remove the other test that reads json and use this approach?